### ISSUE TITLE
[Dev] Merge PR script defaults to the wrong milestone when merging

### ISSUE THREAD

**Issue by raulcd** [apache/arrow#36634]
### Describe the bug, including details regarding any error messages, version, and platform.

I saw that lots of PRs were merged with the wrong milestone (13.0.0) now that the maintenance branch has been created it should default to 14.0.0 but it is not as seen:
```
Would you like to update the associated issue? (y/n): y
Enter fix version [13.0.0]:
```
It does seem that the `12.0.0` and `12.0.1` maintenance branches are also not shown:
```
(Pdb) pr.maintenance_branches
['maint-0.11.x', 'maint-0.12.x', 'maint-0.14.x', 'maint-0.15.x', 'maint-0.17.x', 'maint-1.0.x', 'maint-3.0.x', 'maint-4.0.x', 'maint-6.0.x', 'maint-7.0.x', 'maint-7.0.1', 'maint-8.0.x', 'maint-9.0.0', 'maint-10.0.x', 'maint-10.0.0', 'maint-10.0.1', 'maint-11.0.0']
(Pdb) 
```
It might be due to pagination when requesting branches:
```
    def get_branches(self):
        return get_json("%s/branches" % (self.github_api),
                        headers=self.headers)
```

### Component(s)

Developer Tools

**Comment by kou** [comment]
Yes. Could you remove them?

> [decimal256](https://github.com/apache/arrow/tree/decimal256)

This was merged by #8475

> * [new-contrib-language](https://github.com/apache/arrow/tree/new-contrib-language)

This was merged by #13217 and I removed it now.

> [flight-sql-jdbc](https://github.com/apache/arrow/tree/flight-sql-jdbc)

This was merged by #13800.

> [ibis-remove-string-contains](https://github.com/apache/arrow/tree/ibis-remove-string-contains)

This is garbage.

> [dom/aray](https://github.com/apache/arrow/tree/dom/aray)

This was merged by #35772 and I removed it now.

**Comment by raulcd** [comment]
Thanks @kou , I've removed:
[decimal256](https://github.com/apache/arrow/tree/decimal256)
[flight-sql-jdbc](https://github.com/apache/arrow/tree/flight-sql-jdbc)
[ibis-remove-string-contains](https://github.com/apache/arrow/tree/ibis-remove-string-contains)
and all the release candidate branches for old releases.
